src/gc-arena-derive/src/lib.rs

```diff
@@ -33,15 +34,32 @@ fn collect_derive(mut s: synstructure::Structure) -> TokenStream {
                         panic!("`Collect` mode was already specified with `#[collect({})]`, cannot specify twice", prev_mode_str);
                     }
 
-                    if let Some(syn::NestedMeta::Meta(syn::Meta::Path(path))) = nested.first() {
-                        if path.is_ident("require_static") {
-                            mode = Some(Mode::RequireStatic);
-                        } else if path.is_ident("no_drop") {
-                            mode = Some(Mode::NoDrop);
-                        } else if path.is_ident("unsafe_drop") {
-                            mode = Some(Mode::UnsafeDrop);
-                        } else {
-                            panic!("`#[collect]` requires one of: \"require_static\", \"no_drop\", or \"unsafe_drop\" as an argument");
+                    for nested in nested {
+                        match nested {
+                            syn::NestedMeta::Meta(syn::Meta::Path(path)) => {
+                                if path.is_ident("require_static") {
+                                    mode = Some(Mode::RequireStatic);
+                                } else if path.is_ident("no_drop") {
+                                    mode = Some(Mode::NoDrop);
+                                } else if path.is_ident("unsafe_drop") {
+                                    mode = Some(Mode::UnsafeDrop);
+                                } else {
+                                    panic!("`#[collect]` requires one of: \"require_static\", \"no_drop\", or \"unsafe_drop\" as an argument");
+                                }
+                            }
+                            syn::NestedMeta::Meta(syn::Meta::NameValue(value)) => {
+                                if value.path.is_ident("bound") {
+                                    match value.lit {
+                                        syn::Lit::Str(x) => override_bound = Some(x),
+                                        _ => {
+                                            panic!("`#[collect]` bound expression must be a string")
+                                        }
+                                    }
+                                } else {
+                                    panic!("`#[collect]` encountered an unknown nested item");
+                                }
+                            }
+                            _ => panic!("`#[collect]` encountered an unknown nested item"),
```

Comment:
Would it be possible to fix and unify the error messages here message here:
```
"`#[collect]` requires one of: `require_static`, `no_drop`, `unsafe_drop`, or `\"bound = \"\"`"
```
 or something like that?